recipes/mold/1.4.x/conanfile.py

```diff
@@ -0,0 +1,93 @@
+import os
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import copy, get, rmdir
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.scm import Version
+from conan.tools.env import VirtualBuildEnv
+
+class MoldConan(ConanFile):
+    name = "mold"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/rui314/mold/"
+    license = "AGPL-3.0"
+    description = ("mold is a faster drop-in replacement for existing Unix linkers. It is several times faster than the LLVM lld linker")
+    topics = ("mold", "ld", "linkage", "compilation")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_mimalloc": [True, False],
+    }
+    default_options = {
+        "with_mimalloc": False,
+    }
+
+    def validate(self):
+        if self.settings.build_type == "Debug":
+            raise ConanInvalidConfiguration('Mold is a build tool, specify mold:build_type=Release in your build profile, see https://github.com/conan-io/conan-center-index/pull/11536#issuecomment-1195607330')
+        if self.settings.compiler in ["gcc", "clang", "intel-cc"] and self.settings.compiler.libcxx != "libstdc++11":
+            raise ConanInvalidConfiguration('Mold can only be built with libstdc++11; specify mold:compiler.libcxx=libstdc++11 in your build profile')
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f'{self.name} can not be built on {self.settings.os}.')
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "10":
+            raise ConanInvalidConfiguration("GCC version 10 or higher required")
+        if self.settings.compiler in ('clang', 'apple-clang') and Version(self.settings.compiler.version) < "12":
+            raise ConanInvalidConfiguration("Clang version 12 or higher required")
+        if self.settings.compiler == "apple-clang" and "armv8" == self.settings.arch :
+            raise ConanInvalidConfiguration(f'{self.name} is still not supported by Mac M1.')
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
```

Comment:
```suggestion

    def package_id(self):
        del self.info.settings.compiler

```
Compiler should not be part of the package ID: https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md#settings